### ISSUE TITLE
The checkstyle format includes the current directory

### ISSUE THREAD

**Issue by CLL-GTA** [PHP-CS-Fixer/PHP-CS-Fixer#6097]
## Bug report

PHP
```
PHP 7.4.25 (cli) (built: Oct 23 2021 18:05:05) ( NTS )
Copyright (c) The PHP Group
Zend Engine v3.4.0, Copyright (c) Zend Technologies
    with Zend OPcache v7.4.25, Copyright (c), by Zend Technologies
```

php-cs-fixer
```
PHP CS Fixer 3.2.1 Mountains by Fabien Potencier and Dariusz Ruminski
```

I use the `--format=checkstyle` option as I am running the output through `cs2pr` in a GH Action. However `cs2pr` fails to add annotation because the file name in the XML out[ut contains the current directory.

I fixed my GH Action to remove it but I consider this a bug in php-cs-fixer.

To reproduce I have used this file

```php
<?php

class Foo
{

}
```

Run the following command 

```
php-cs-fixer fix --dry-run --using-cache=no --format=checkstyle -vvv ./test.php
```

and the output is

```xml
<?xml version="1.0" encoding="UTF-8"?>
<checkstyle>
  <file name="CareLineLive/test.php">
    <error severity="warning" source="PHP-CS-Fixer.method_argument_space" message="Found violation(s) of type: method_argument_space"/>
  </file>
</checkstyle>
```

I believe the `name` attribute should be `name="test.php"` or maybe `name="./test.php"`.

**Comment by CLL-GTA** [comment]
The issue it's related to the use of `--using-cache=no`.

In `\PhpCsFixer\Console\ConfigurationResolver::getDirectory` the `$path` is `null` and therefore `$absolutePath` is the current working directory, `/Users/giulio/projects/CareLineLive` in my case.

The problem is that `\dirname()` is called later, which will return `/Users/giulio/projects`, which then means the name of the file is `CareLineLive/test.php`.

I have modified the `getDirectory()` method locally and it works. I'm not sure of the implications though

```php
    public function getDirectory(): DirectoryInterface
    {
        if (null === $this->directory) {
            $path = $this->getCacheFile();
            if (null === $path) {
                $this->directory = new Directory($this->cwd);
            } else {
                $filesystem = new Filesystem();

                $absolutePath = $filesystem->isAbsolutePath($path)
                    ? $path
                    : $this->cwd . \DIRECTORY_SEPARATOR . $path;

                $this->directory = new Directory(\dirname($absolutePath));
            }
        }

        return $this->directory;
    }
```

**Comment by troccoli** [comment]
I have created a PR to fix this.

However the first data set in `\PhpCsFixer\Tests\Console\ConfigurationResolverTest::provideGetDirectoryCases` used by `\PhpCsFixer\Tests\Console\ConfigurationResolverTest::testGetDirectory()` explicitly expects `path/my/file`. I have updated the test obviously, but I still wonder if there was another reason why the expected path was `path/my/file`.

**Comment by Wirone** [comment]
Closing as this is not reproducible on `3.16.1-DEV` (current master with unreleased changes):

<img width="1310" alt="image" src="https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/assets/600668/73fa3b64-bfb3-41ec-9dcc-70044058a6e1">

Most probably fixed in #6915.
